wremnants/include/muon_calibration.h

```diff
@@ -450,11 +450,68 @@ using base_t = BiasCorrectionHelper<T>;
     //inherit constructor
     using base_t::base_t;
 
-    RVec<float> operator() (const RVec<float> etas, const RVec<float>& pts, RVec<int> charges) {
+    RVec<float> operator() (const RVec<float>& pts, const RVec<float>& etas) {
         RVec<float> biased_pt(pts.size(), 0.);
-        assert(etas.size() == pts.size() && etas.size() == charges.size());
+        assert(etas.size() == pts.size());
         for (size_t i = 0; i < pts.size(); i++) {
-            biased_pt[i] = BiasCorrectionHelper<T>::operator()(etas[i], pts[i], charges[i]);
+            biased_pt[i] = BiasCorrectionHelper<T>::operator()(pts[i], etas[i]);
+        }
+
+        return biased_pt;
+    }
+};
+
+template <typename T>
+class SmearingHelperBase {
+
+public:
+    SmearingHelperBase(T&& corrections) :
+        smearingHist_(std::make_shared<const T>(std::move(corrections))) {
+            myRndGen = new TRandom3();
+            myRndGen->SetSeed(1); // not 0 because seed 0 has a special meaning
+        }
+    
+    // helper for bin lookup which implements the compile-time loop over axes
+    template<typename... Xs, std::size_t... Idxs>
```

Comment:
Couldn't this more or less inherit from the other helper (or make a new common class) to reduce the code duplication?